app/addons/documents/views-doceditor.js

```diff
@@ -50,16 +49,20 @@ function (app, FauxtonAPI, Components, Documents, Databases, prettify) {
 
       this.$('#_rev').val(docRev);
 
-      $form.ajaxSubmit({
-        url: this.model.url(),
-        type: 'POST',
-        beforeSend: this.beforeSend,
-        uploadProgress: this.uploadProgress,
+      $.ajax({
+        url: this.model.url() + '/' + file.name + '?rev=' + docRev,
+        xhrFields: {
+          withCredentials: true
+        },
+        type: 'PUT',
+        data: file,
+        processData: false,
+        contentType: file.type,
         success: this.success,
         error: function (resp) {
-          console.log('ERR on upload', resp);
-          return that.set_error_msg('Could not upload document: ' + JSON.parse(resp.responseText).reason);
-        }
+          this.set_error_msg('Could not upload document: ' + JSON.parse(resp.responseText).reason);
```

Comment:
I think "Could not upload attachment" would be better message here since actually request uploads an attached file, not document. Otherwise this message creates logically conflict with the error on L43 when you can update document and add attachment at the same time.